packages/inter-protocol/test/auction/test-scheduler.js

```diff
@@ -0,0 +1,521 @@
+import { test } from '@agoric/zoe/tools/prepare-test-env-ava.js';
+import { buildManualTimer } from '@agoric/swingset-vat/tools/manual-timer.js';
+import { TimeMath } from '@agoric/time';
+import { setupZCFTest } from '@agoric/zoe/test/unitTests/zcf/setupZcfTest.js';
+import { eventLoopIteration } from '@agoric/zoe/tools/eventLoopIteration.js';
+
+import { makeScheduler } from '../../src/auction/scheduler.js';
+import {
+  makeAuctioneerParamManager,
+  makeAuctioneerParams,
+} from '../../src/auction/params.js';
+import {
+  getInvitation,
+  makeDefaultParams,
+  makeFakeAuctioneer,
+  makePublisherFromFakes,
+  setUpInstallations,
+} from './tools.js';
+
+/** @typedef {import('@agoric/time/src/types').TimerService} TimerService */
+
+test('schedule start to finish', async t => {
+  const { zoe } = await setupZCFTest();
+  const installations = await setUpInstallations(zoe);
+  /** @type {TimerService & { advanceTo: (when: Timestamp) => void; }} */
+  const timer = buildManualTimer();
+  const timerBrand = await timer.getTimerBrand();
+
+  const fakeAuctioneer = makeFakeAuctioneer();
+  const { fakeInvitationPayment } = await getInvitation(zoe, installations);
+  const publisherKit = makePublisherFromFakes();
+
+  let defaultParams = makeDefaultParams(fakeInvitationPayment, timerBrand);
+  defaultParams = {
+    ...defaultParams,
+    auctionStartDelay: 1n,
+    startFreq: 10n,
+    priceLockPeriod: 5n,
+  };
+  const params = makeAuctioneerParams(defaultParams);
+  const params2 = {};
+  for (const key of Object.keys(params)) {
+    const { value } = params[key];
+    params2[key] = value;
+  }
+
+  let now = 127n;
+  await timer.advanceTo(now);
```

Comment:
if you make `advanceTo` return its argument all the "set now and advance' pairs could be one line:
```suggestion
  let now = await timer.advanceTo(127n);
```

Reply:
Done. There weren't as many two-line versions as it seemed.